### DISCUSSION TITLE
Issue renewing Njalla cert

### DISCUSSION THREAD

**Discussion by chaosyoga** [go-acme/lego#2347] [Q&A] [answered]
Hi, I was using https://github.com/chaptergy/certbot-dns-njalla/blob/main/setup.py until recently to renew my Njalla certs, but they failed to update and now I found your solution, but am having some issues. 

I installed lego and used the bash example provided:
```
NJALLA_TOKEN=xxxxxxxxxxxxxxxxxxxxxxxxxx \
lego --email you@example.com --dns njalla -d '*.example.com' -d example.com run
```
This gives me `unrecognized DNS provider: njalla`, I'm using protonmail email for the cert if that makes a difference. 

Am I able to renew the old cert this way, or do I have to start from scratch?

**Comment by ldez** [comment]
Hello @chaosyoga,

can you provide the version of lego you are using?

```
lego --version
```

**Reply by chaosyoga** [reply]
`lego version dev linux/amd64`

**Reply by ldez** [reply]
how did you install it?

FYI `dev` is not a real version.

**Reply by chaosyoga** [reply]
Installed via apt.

**Reply by ldez** [reply, marked as answer]
We are not the maintainers of deb packages, and they are all out-of-date.

I recommend installing from our binaries: https://github.com/go-acme/lego/releases

**Reply by chaosyoga** [reply]
I removed it and installed the snap, still get the same version.

**Reply by ldez** [reply]
**`dev` is not a real version.**

Snap is also out-of-date but less than the deb packages.

**Reply by chaosyoga** [reply]
I understand that dev is not a real version, which is why I'm mentioning its same as apt.

I downloaded and unziped the linux tar, now how do I install?

**Reply by ldez** [reply]
Inside the tar you have 3 files:
```
.
├── CHANGELOG.md
├── LICENSE
└── lego
```

`lego` is the executable file.

You can:
- run the executable file: `./lego`
- copy the executable file inside a directory referenced inside your `PATH` (ex: `/usr/local/bin`) and run it `lego`.


> I understand that dev is not a real version, which is why I'm mentioning its same as apt.

Maybe I was not clear: `dev` is not a specific version. The real version of lego inside apt and snap are not the same.
The problem is related to the fact that the packager has not handled the version as expected inside deb and snap.

**Reply by chaosyoga** [reply]
Ok, should I run from scratch and install a new cert, or can I just renew my previous cert now?

**Reply by ldez** [reply]
lego is not a drop-in replacement for Certbot.
FYI lego existed before Certbot.

So the easiest way is to "run from scratch".

**Reply by ldez** [reply]
give me the command you are trying to run (redact the personal information)

**Reply by chaosyoga** [reply]
Ok, making progress. It asked to accept Tos, but now I get `Invalid token header. Token string should not contain spaces.`

**Reply by ldez** [reply]
You have spaces inside your token.

**Reply by chaosyoga** [reply]
I don't, but there was before lego and after \ so that fixed it.

Now getting: `Invalid token`

I created a new token in Njalla and tried that too, same error.

**Reply by ldez** [reply]
give me the command you are trying to run (redact the personal information)

**Comment by chaosyoga** [comment]
```
NJALLA_TOKEN=XXXXXXXXXX\sudo lego --email my@email.com --dns njalla -d '*.my.domain' -d my.domain run
```

**Reply by ldez** [reply]
try this:

```bash
export NJALLA_TOKEN="XXXXXXXXXX"
sudo lego --email my@email.com --dns njalla -d '*.my.domain' -d my.domain run
```

**Reply by chaosyoga** [reply]
`njalla: some credentials information are missing: NJALLA_TOKEN`

**Reply by ldez** [reply]
FYI you don't need to run with `sudo`, the DNS challenge inside lego doesn't require root.

**Reply by ldez** [reply]
Do that:

```bash
export NJALLA_TOKEN="XXXXXXXXXX"
```

and 

```bash
echo $NJALLA_TOKEN
```

```bash
sudo echo $NJALLA_TOKEN
```

Do you see the token?

**Reply by chaosyoga** [reply]
Removing sudo did the trick:

```
export NJALLA_TOKEN="XXXXXXXXXX"
lego --email my@email.com --dns njalla -d '*.my.domain' -d my.domain run
```

Now I see `Server responded with a certificate.`

**Reply by chaosyoga** [reply]
What now? My site still shows `ERR_CERT_DATE_INVALID`

**Reply by ldez** [reply]
You have to put the certificates in the right place
I don't know your infrastructure.

**Reply by chaosyoga** [reply]
Where is the newly generated lego cert located now?

Currently I'm using Certbot with this plugin: https://github.com/chaptergy/certbot-dns-njalla

**Reply by ldez** [reply]
inside the directory `.lego`

**Reply by chaosyoga** [reply]
Should I uninstall Certbot now? How can I use these new certs by lego?

**Reply by ldez** [reply]
> Should I uninstall Certbot now?

You can if everything is working, but you don't need it.

> How can I use these new certs by lego?

I don't understand your question.

As I said, I don't know your infrastructure, so I can just say that you should place your certificates where the certificates should be placed inside your infrastructure.

**Reply by chaosyoga** [reply]
Currently I see the old certs in `/etc/letsencrypt/live/domain` in `.pem` format, should move them there?

**Reply by ldez** [reply]
I don't know your infrastructure, I don't know how your infrastructure works, so I cannot answer to that.

I can only answer with: probably yes but I don't know.
